src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -195,27 +195,145 @@ class WorkflowManagerActor(dataAccess: DataAccess, backend: Backend) extends Act
                                 callInputs: Traversable[Symbol],
                                 callOutputs: Traversable[Symbol],
-                                jobInfos: Map[ExecutionDatabaseKey, Any]): Map[FullyQualifiedName, Seq[CallMetadata]] = {
-
-    // add straightforward queries for call inputs and outputs
-    // navigate the keys of the map and indices of their arrays to effectively unify the data
-    // into CallMetadata.
-    // figure out how to get job IDs and maybe on which backend the call executed
-    standardStreamsMap map { case (key, seqOfStreams) =>
-      key -> seqOfStreams.map { streams =>
-        CallMetadata(
-          inputs = Map("input_key" -> "input_value"),
-          status = "UnknownStatus",
-          backend = Option("UnknownBackend"),
-          outputs = Option(Map("output_key" -> "output_value")),
-          start = Option(new DateTime()),
-          end = Option(new DateTime()),
-          jobId = Option("COMPLETELY-MADE-UP-ID"),
-          rc = Option(0),
-          stdout = Option(streams.stdout),
-          stderr = Option(streams.stderr))
+                                jobMap: Map[ExecutionDatabaseKey, Any]): Map[FullyQualifiedName, Seq[CallMetadata]] = {
+
+    def filterCollectorSymbols(symbols: Traversable[Symbol]): Traversable[Symbol] = {
+      // Squash duplicate ExecutionDatabaseKeys.
+      val databaseKeys = symbols map { s => ExecutionDatabaseKey(s.scope, s.index.toIndex) } toSet
+      // Look for FQNs with more than one ExecutionDatabaseKey per FQN.
+      val fqnsWithCollectors = databaseKeys.groupBy(_.fqn).collect ({ case (fqn, keys) if keys.size > 1 => fqn }).toSet
+      // We know which FQNs with None indexes correspond to collectors, so filter matching symbols.
+      symbols.filterNot { s => s.index.toIndex.isEmpty && fqnsWithCollectors.contains(s.scope) }
+    }
+
+    // Collects up all the data for a single CallMetadata in the easiest-to-assemble format, but
+    // not quite the format required for the metadata REST endpoint.
+    case class AssembledCallMetadata(key: ExecutionDatabaseKey,
+                                     execution: Execution,
+                                     inputs: Traversable[Symbol] = Seq.empty,
+                                     outputs: Option[Traversable[Symbol]] = None,
+                                     streams: Option[StdoutStderr] = None,
+                                     backend: Option[String] = None,
+                                     jobId: Option[String] = None,
+                                     backendStatus: Option[String] = None)
+
+    val executionKeys = executions map { x => ExecutionDatabaseKey(x.callFqn, x.index.toIndex) }
+
+    // Define a sequence of ExecutionMap transformer functions and fold a zero Map across them.
+    type ExecutionMap = Map[ExecutionDatabaseKey, AssembledCallMetadata]
+    type ExecutionMapTransformer = ExecutionMap => ExecutionMap
```

Comment:
The function values below are the smaller units.  They need to agree on a type as they are put into a `Seq` for `foldLeft` to build the canonical `ExecutionMap`, so these definitions are here.

I'll add more documentation, this isn't as clear as it should be.